vite-plugin-cjs-interop/src/index.ts

```diff
@@ -37,9 +37,18 @@ export function cjsInterop(options: CjsInteropOptions): Plugin {
 	const { client = false, apply = "both" } = options;
 
 	let sourcemaps = false;
+
+	const matchedDependencies: Record<string, boolean> = {};
+
 	const matchesDependencies = (value: string) => {
-		return dependencies.some((dependency) => minimatch(value, dependency));
+		if (!(value in matchedDependencies)) {
+			matchedDependencies[value] = dependencies.some((dependency) =>
+				minimatch(value, dependency),
+			);
+		}
+		return matchedDependencies[value];
```

Comment:
This is an unrelated optimization; Some memoizing here speeds the plugin to a good amount.